### PR TITLE
[RS-1843] Operator changes to introduce configurable Snort rules

### PR DESCRIPTION
**Pull request by bartolini**
Original JIRA ticket: https://tigera.atlassian.net/browse/RS-1843

### CODE REVIEW

File: pkg/render/intrusiondetection/dpi/dpi_test.go [hunk 1]
```diff
@@ -208,6 +210,12 @@ var _ = Describe("DPI rendering tests", func() {
 		dpiCertSecret, err = certificateManager.GetOrCreateKeyPair(cli, render.DPITLSSecretName, common.OperatorNamespace(), []string{""})
 		Expect(err).NotTo(HaveOccurred())
 
+		snortRules, err := embed.AsConfigMap(
+			applicationlayer.ModSecurityRulesetConfigMapName,
```
Comment (electricjesus) [inline pkg/render/intrusiondetection/dpi/dpi_test.go:214]: should this be `dpi.DeepPacketInspectionSnortRulesCmName`?

File: pkg/render/intrusiondetection/dpi/dpi_test.go [hunk 2]
```diff
@@ -208,6 +210,12 @@ var _ = Describe("DPI rendering tests", func() {
 		dpiCertSecret, err = certificateManager.GetOrCreateKeyPair(cli, render.DPITLSSecretName, common.OperatorNamespace(), []string{""})
 		Expect(err).NotTo(HaveOccurred())
 
+		snortRules, err := embed.AsConfigMap(
```
Comment (electricjesus) [inline pkg/render/intrusiondetection/dpi/dpi_test.go:213]: If you use the applicationlayer embed package, this means that your configmap contents is going to be whatever is in [this](https://github.com/tigera/operator/tree/master/pkg/render/applicationlayer/embed/coreruleset) directory. is this OK?

File: pkg/render/intrusiondetection/dpi/dpi_test.go [hunk 3]
```diff
@@ -208,6 +210,12 @@ var _ = Describe("DPI rendering tests", func() {
...
+			applicationlayer.ModSecurityRulesetConfigMapName,
+			common.OperatorNamespace(),
+		)
+		Expect(err).To(BeNil())
```
Comment (electricjesus) [inline pkg/render/intrusiondetection/dpi/dpi_test.go:217]: I think this whole lot can just be replaced with.. let me know if this is incorrect 

```suggestion
		snortRules := &corev1.ConfigMap{
			TypeMeta: metav1.TypeMeta{Kind: "ConfigMap", APIVersion: "v1"},
			ObjectMeta: metav1.ObjectMeta{
				Name:      dpi.DeepPacketInspectionSnortRulesCmName,
				Namespace: common.OperatorNamespace(),
			},
		}
```

File: pkg/render/intrusiondetection/dpi/dpi_test.go (hunk 2 above)
Comment (bartolini) [reply pkg/render/intrusiondetection/dpi/dpi_test.go]: Thanks, changed as per previous comment.

File: pkg/render/intrusiondetection/dpi/dpi_test.go (hunk 1 above)
Comment (bartolini) [reply pkg/render/intrusiondetection/dpi/dpi_test.go]: That's right , this is a copy-paste error. Corrected.

**Review by electricjesus — APPROVED** [review]
LGTM

File: pkg/controller/intrusiondetection/intrusiondetection_controller.go [hunk 4]
```diff
@@ -518,6 +522,25 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 			return reconcile.Result{}, err
 		}
 
+		customSnortRulesCm := new(corev1.ConfigMap)
```
Comment (rene-dekker) [inline pkg/controller/intrusiondetection/intrusiondetection_controller.go:525]: ConfigMaps can store up to 1 MiB of data, while the set of snort rules may quite possibly be larger than this.

**Pull request closed without merging**